### PR TITLE
Ensure tab panels are labelled and have `tabindex` set to 0.

### PR DESCRIPTION
**Pull request by khiga8**
## What
- Updates panel slot in `Navigation::Tab` to ensure there's a label. For ease, if `id` is set on the tab, we auto-label the panel with the tab text using `aria-labelledby`. Consumers may alternatively pass in an `aria-label`.
- Sets panel `tabindex="0"`. 

## Why
According to [WAI-ARIA Roles, States, and Properties for tabs](https://www.w3.org/TR/wai-aria-practices/#wai-aria-roles-states-and-properties-20) and [Example of Tabs](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html), tab panels should be labelled. Additionally, tab panel should have a tabindex set to 0.

## Notes

- There are a lot of updates I want to make with the tab components but I also want to keep the diffs small for ease of review. Ultimately, I think we need to move towards separating the current `UnderlineNav` into `UnderlinePanelNav` and `UnderlinePageNav`. Same thing with `TabNav`. [Related issue](https://github.com/primer/view_components/issues/680)

### CODE REVIEW

**Comment by manuelpuyol** [comment]
I'm pretty sure that `tab-container` should manage that for us 🤔

**Comment by khiga8** [comment]
> I'm pretty sure that `tab-container` should manage that for us 🤔

@manuelpuyol  Nope :( 

The labeling is recommended in the [example markup](https://github.com/github/tab-container-element#usage) for `tab-container`.

**Comment by manuelpuyol** [comment]
> > I'm pretty sure that `tab-container` should manage that for us 🤔
> 
> @manuelpuyol Nope :(
> 
> The labeling is recommended in the [example markup](https://github.com/github/tab-container-element#usage) for `tab-container`.

what about the `tabindex`? I think the element sets it after selecting a tab

**Comment by khiga8** [comment]
> > > I'm pretty sure that `tab-container` should manage that for us 🤔
> > 
> > 
> > @manuelpuyol Nope :(
> > The labeling is recommended in the [example markup](https://github.com/github/tab-container-element#usage) for `tab-container`.
> 
> what about the `tabindex`? I think the element sets it after selecting a tab

Hrmm yeah. The `tabindex` gets set if tab is selected but we should make sure it's set always.

File: app/components/primer/navigation/tab_component.rb [hunk 1]
```diff
@@ -18,10 +18,22 @@ class TabComponent < Primer::Component
...
       # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
       renders_one :panel, lambda { |**system_arguments|
+        return unless @with_panel
+
         system_arguments[:tag] = :div
         system_arguments[:role] ||= :tabpanel
+        system_arguments[:tabindex] = 0
         system_arguments[:hidden] = true unless @selected
 
+        label_present = system_arguments[:"aria-label"] || system_arguments.dig(:aria, :label) || system_arguments[:"aria-labelledby"] || system_arguments.dig(:aria, :labelledby)
```
Comment (manuelpuyol) [inline app/components/primer/navigation/tab_component.rb:28]: I think it is time to write helpers to get aria values, this double check is getting propagated everywhere and it's not great 😅 

We could do something like this in `Primer::Component`:
```rb
def aria(val, system_arguments)
  system_arguments[:"aria-#{val}"] || system_arguments.dig(:aria, val)
end
```

Comment (khiga8) [reply app/components/primer/navigation/tab_component.rb]: great idea!

Comment (khiga8) [reply app/components/primer/navigation/tab_component.rb]: I may add a linter in a followup.

File: app/components/primer/navigation/tab_component.rb [hunk 2]
```diff
@@ -18,10 +18,22 @@ class TabComponent < Primer::Component
...
       # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
       renders_one :panel, lambda { |**system_arguments|
+        return unless @with_panel
+
         system_arguments[:tag] = :div
         system_arguments[:role] ||= :tabpanel
+        system_arguments[:tabindex] = 0
         system_arguments[:hidden] = true unless @selected
 
+        label_present = aria("label", system_arguments) || system_arguments[:"aria-labelledby"] || system_arguments.dig(:aria, :labelledby)
```
Comment (manuelpuyol) [inline app/components/primer/navigation/tab_component.rb:28]:
```suggestion
        label_present = aria("label", system_arguments) || aria("labelledby", system_arguments)
```

**Review by manuelpuyol — APPROVED** [review]

**Pull request merged**